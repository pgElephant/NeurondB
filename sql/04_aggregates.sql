-- Test aggregate functions
CREATE TEMP TABLE test_vectors (id int, v vector);
INSERT INTO test_vectors VALUES 
    (1, '[1.0, 0.0]'),
    (2, '[0.0, 1.0]'),
    (3, '[1.0, 1.0]'),
    (4, '[2.0, 2.0]');

-- Test vector average
SELECT vector_avg(v) FROM test_vectors;

-- Test vector sum
SELECT vector_sum(v) FROM test_vectors;

-- Test with WHERE clause
SELECT vector_avg(v) FROM test_vectors WHERE id <= 2;

