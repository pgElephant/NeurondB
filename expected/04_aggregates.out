-- Test aggregate functions
CREATE TEMP TABLE test_vectors (id int, v vector);
ERROR:  type "vector" does not exist
LINE 1: CREATE TEMP TABLE test_vectors (id int, v vector);
                                                  ^
INSERT INTO test_vectors VALUES 
    (1, '[1.0, 0.0]'),
    (2, '[0.0, 1.0]'),
    (3, '[1.0, 1.0]'),
    (4, '[2.0, 2.0]');
ERROR:  relation "test_vectors" does not exist
LINE 1: INSERT INTO test_vectors VALUES 
                    ^
-- Test vector average
SELECT vector_avg(v) FROM test_vectors;
ERROR:  relation "test_vectors" does not exist
LINE 1: SELECT vector_avg(v) FROM test_vectors;
                                  ^
-- Test vector sum
SELECT vector_sum(v) FROM test_vectors;
ERROR:  relation "test_vectors" does not exist
LINE 1: SELECT vector_sum(v) FROM test_vectors;
                                  ^
-- Test with WHERE clause
SELECT vector_avg(v) FROM test_vectors WHERE id <= 2;
ERROR:  relation "test_vectors" does not exist
LINE 1: SELECT vector_avg(v) FROM test_vectors WHERE id <= 2;
                                  ^
